infrastructure/quick-deploy/aws/all/monitoring.tf

```diff
@@ -0,0 +1,226 @@
+  depends_on  = [module.metrics_exporter]
+}
+
+# Prometheus
+module "prometheus" {
+  source                         = "../../../modules/monitoring/prometheus"
+  namespace                      = local.namespace
+  service_type                   = var.prometheus.service_type
+  node_selector                  = var.prometheus.node_selector
+  metrics_exporter_url           = "${module.metrics_exporter.host}:${module.metrics_exporter.port}"
+  partition_metrics_exporter_url = length(module.partition_metrics_exporter) == 1 ? "${module.partition_metrics_exporter[0].host}:${module.partition_metrics_exporter[0].port}" : null
+  docker_image = {
+    image              = local.ecr_images["${var.prometheus.image_name}:${var.prometheus.image_tag}"].image
+    tag                = local.ecr_images["${var.prometheus.image_name}:${var.prometheus.image_tag}"].tag
+    image_pull_secrets = var.prometheus.pull_secrets
+  }
+  working_dir = "${path.root}/../../.."
+}
+
+# Grafana
+module "grafana" {
+  count          = var.grafana != null ? 1 : 0
+  source         = "../../../modules/monitoring/grafana"
+  namespace      = local.namespace
+  service_type   = var.grafana.service_type
+  port           = var.grafana.port
+  node_selector  = var.grafana.node_selector
+  prometheus_url = module.prometheus.url
+  docker_image = {
+    image              = local.ecr_images["${var.grafana.image_name}:${var.grafana.image_tag}"].image
+    tag                = local.ecr_images["${var.grafana.image_name}:${var.grafana.image_tag}"].tag
+    image_pull_secrets = var.grafana.pull_secrets
+  }
+  working_dir    = "${path.root}/../../.."
+  authentication = var.grafana.authentication
+}
+
+# CloudWatch
+module "cloudwatch" {
+  count             = var.cloudwatch != null ? 1 : 0
+  source            = "../../../modules/aws/cloudwatch-log-group"
+  name              = local.cloudwatch_log_group_name
+  kms_key_id        = local.kms_key
+  retention_in_days = var.cloudwatch.retention_in_days
+  tags              = local.tags
+}
+
+# Fluent-bit
+module "fluent_bit" {
+  source        = "../../../modules/monitoring/fluent-bit"
+  namespace     = local.namespace
+  node_selector = var.fluent_bit.node_selector
+  fluent_bit = {
+    container_name     = "fluent-bit"
+    image              = local.ecr_images["${var.fluent_bit.image_name}:${var.fluent_bit.image_tag}"].image
+    tag                = local.ecr_images["${var.fluent_bit.image_name}:${var.fluent_bit.image_tag}"].tag
+    image_pull_secrets = var.fluent_bit.pull_secrets
+    is_daemonset       = var.fluent_bit.is_daemonset
+    http_server        = (var.fluent_bit.http_port == 0 ? "Off" : "On")
+    http_port          = (var.fluent_bit.http_port == 0 ? "" : tostring(var.fluent_bit.http_port))
+    read_from_head     = (var.fluent_bit.read_from_head ? "On" : "Off")
+    read_from_tail     = (var.fluent_bit.read_from_head ? "Off" : "On")
+  }
+  seq = length(module.seq) != 0 ? {
+    host    = module.seq.0.host
```

Comment:
homogenize array accesses, sometimes we use `module.modulename[0].property` and in others `module.modulename.0.property`

Reply:
As far as I remember, it is recommended to use `[0]`, but in ArmoniK, mostly `.0` is used. I can use `.0`, but I would prefer use `[0]`.

Reply:
I found what I had in mind. It is not about the `.0` per se, but `.*`: https://developer.hashicorp.com/terraform/language/expressions/splat#legacy-attribute-only-splat-expressions

The logical extension would be to use neither `.*`, nor `.*` to be consistent.

Reply:
I also prefer `[0]`,  but we can fix that in another PR